src/OpenTelemetry.AutoInstrumentation/Loading/LazyInstrumentationLoader.cs

```diff
@@ -0,0 +1,96 @@
+// <copyright file="LazyInstrumentationLoader.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#if NETCOREAPP3_1_OR_GREATER
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+using OpenTelemetry.AutoInstrumentation.Logging;
+
+namespace OpenTelemetry.AutoInstrumentation.Loading;
+
+internal class LazyInstrumentationLoader : ILifespanManager, IDisposable
```

Comment:
Can you document what is this class responsible for?

Reply:
6697d7a532eff4e792cf129d88c941b8085d91f9